### ISSUE TITLE
[Bug] Replaces Own `Validator` type with `PropTypes.Validator`

### ISSUE THREAD

**Issue by daniel-nagy** [eps1lon/types-react-codemod#393]
Hi,

I just used the `types-react-codemod@latest preset-19` per the [TypeScript Changes](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#typescript-changes) in the React 19 upgrade guide and it replaced my own `Validator` type in one of my files with `PropTypes.Validator`.

Not a big deal but thought I would report it. Thanks for the codemod!

**Comment by eps1lon** [comment]
Yeah I cheated here a bit and just looked for `Validator` without actually checking if it comes from `react`. Would love to see a fix from somebody but I won't have time to prioritize this.

**Comment by NianJiuZst** [comment]
> Yeah I cheated here a bit and just looked for `Validator` without actually checking if it comes from `react`. Would love to see a fix from somebody but I won't have time to prioritize this.

Could you assign the issue to me？

**Comment by NianJiuZst** [comment]
@eps1lon

**Comment by eps1lon** [comment]
You can start working on this issue without being assigned.
